components/char_collection/src/char_collection.rs

```diff
@@ -0,0 +1,828 @@
+// Copyright 2019 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+use std::{
+    boxed::Box,
+    char,
+    clone::Clone,
+    cmp::Ordering,
+    convert::Into,
+    error::Error,
+    hash::{Hash, Hasher},
+    iter::Iterator,
+    ops::Range,
+    vec::Vec,
+};
+
+#[derive(Copy, Clone, Debug, Eq)]
+pub struct CharRange {
+    low: char,
+    high: char,
+}
+
+impl CharRange {
+    // open_right
+    // would we want this to return a Option next time?
+    pub fn open_right(low: char, high: char) -> CharRange {
+        // nothing happens if this fails
+        let high: char = char::from_u32(high as u32 - 1).unwrap();
+        CharRange { low, high }
+    }
```

Comment:
What do you mean `nothing`? This would panic.

Overall, in Rust you could define that as a trait `CharRangeType` and then `impl CharRangeType for Range<char>`.

I think it would be more canonical, but I'll ask Manish to take a look